.github/workflows/doc-builder.yml

```diff
@@ -16,4 +16,5 @@ jobs:
         with:
           python-version: 3.x
       - run: pip install mkdocs-material==8.2.9
+      - run: pip install mkdocs-awesome-pages-plugin
```

Comment:
I have added a specific version to avoid future breaking changes. Good callout.
I don't believe we need to add this to THIRD_PARTY_LICENSES as we are not shipping it to the customers. This is used in our build system. @normj any thoughts here?

Reply:
Per the internal wiki, if this is purely Python that runs during the doc build then I agree it's okay to omit, but if there are any static assets that are part of the doc site then I think we should include it.
> Note that artifacts, like CSS or JavaScript, that go to a customer's browser are distributed.